spec/accept_language/matcher_spec.rb

```diff
@@ -36,6 +37,34 @@
 
       it { expect(best_match).to be_nil }
     end
+
+    context "when primary_fallback is in true" do
+      let(:primary_fallback) { true }
+
+      context "when Chinese is available" do
+        let(:available_langtag) { "zh" }
+
+        it { expect(best_match).to eq "zh" }
```

Comment:
This is the case that's being handle by this change

Reply:
Hi @cyril thanks for getting back, but I think that maybe I'm missing something, because with the current implementation (not the proposed one in this PR) this test case fails, so not sure if I'm understanding correctly 🤔 

The idea is that if the language header is requesting `zh-TW` and there is `zh` language available this should match if the param `primary_fallback` is set on `true`... if not it will return `nil` as it's doing right now, is this what you where referring?

Reply:
Hi @cyril thanks for getting back, but I think that maybe I'm missing something, because with the current implementation (not the proposed one in this PR) this test case fails, so not sure if I'm understanding correctly 🤔

The idea is that if the language header is requesting `zh-TW` and there is `zh` language available this should match if the param `primary_fallback` is set on `true`... if not it will return `nil` as it's doing right now, is this what you where referring?

Maybe I missed something... but I was not aware of the RFC 2616 specification, but since you mentioned this test case shows that it is not, or I'm misunderstanding it?

Reply:
Now I follow! thanks for the detail explanation! I also agree with your statements, my case is somehow particular and I rather handle as that and conform with the RFC.
Thanks!

Reply:
You're welcome! Love seeing your interest in language accessibility 🙂